code/modules/client/preference/loadout/gear_tweaks.dm

```diff
@@ -10,102 +15,51 @@
 /datum/gear_tweak/proc/tweak_gear_data(metadata, datum/gear_data)
 	return
 
-/datum/gear_tweak/proc/tweak_item(obj/item/I, metadata)
+/datum/gear_tweak/proc/tweak_item(obj/item/gear, metadata)
 	return
 
-/*
-* Color adjustment
-*/
-
+// MARK: Color
 /datum/gear_tweak/color
+	display_type = "Color"
+	fa_icon = "palette"
+	info = "Recolorable"
 	var/list/valid_colors
 
 /datum/gear_tweak/color/New(list/colors)
 	valid_colors = colors
 	..()
 
-/datum/gear_tweak/color/get_contents(metadata)
-	return "Color: <font color='[metadata]'>&#9899;</font>"
-
 /datum/gear_tweak/color/get_default()
 	return valid_colors ? valid_colors[1] : COLOR_GRAY
 
 /datum/gear_tweak/color/get_metadata(user, metadata)
 	if(valid_colors)
-		return input(user, "Choose an item color.", "Character Preference", metadata) as null|anything in valid_colors
-	return input(user, "Choose an item color.", "Global Preference", metadata) as color|null
+		return tgui_input_list(user, "Choose an item color.", "Recolor Gear", valid_colors, metadata)
+	return input(user, "Choose an item color.", "Recolor Gear", metadata) as color|null
 
-/datum/gear_tweak/color/tweak_item(obj/item/I, metadata)
-	if(valid_colors && !(metadata in valid_colors))
+/datum/gear_tweak/color/tweak_item(obj/item/gear, metadata)
+	if((valid_colors && !(metadata in valid_colors)) || !metadata)
 		return
-	I.color = metadata
 
-/*
-* Path adjustment
-*/
+	gear.color = metadata
 
-/datum/gear_tweak/path
-	var/list/valid_paths
+// MARK: Rename
+/datum/gear_tweak/rename
+	display_type = "Name"
+	fa_icon = "edit"
+	info = "Renameable"
 
-/datum/gear_tweak/path/New(list/paths)
-	valid_paths = paths
-	..()
+/datum/gear_tweak/rename/get_default()
+	return ""
 
-/datum/gear_tweak/path/get_contents(metadata)
-	return "Type: [metadata]"
+/datum/gear_tweak/rename/get_metadata(user, metadata)
+	var/new_name = tgui_input_text(user, "Rename an object. Enter empty line for stock name", "Rename Gear", metadata, MAX_NAME_LEN)
+	if(isnull(new_name))
```

Comment:
Will isnull() work properly with an empty line, or do you want to just check truthiness here?

Reply:
Works properly, it just prevent clearing name when closing window (and pressing Cancel button as i remember)